src/header/internals/cell.rs

```diff
@@ -87,6 +87,20 @@ impl<V: ?Sized + Any + 'static> PtrMapCell<V> {
     }
 
     #[inline]
+    pub fn into_value(self, key: TypeId) -> Option<Box<V>> {
+        let mut map = unsafe { self.0.into_inner() };
```

Comment:
`UnsafeCell::get` is not unsafe, because it just returns a raw pointer. You can toss those around freely. However, **dereferencing** it is unsafe, which is needed to turn it into a safe pointer (the other way is to use mem::transmute, but that's also unsafe).

The extra caveats for `into_inner` are the same as using the pointer from `UnsafeCell::get`: accessing it is not synchronized over threads, but that is fine for us here, because of the mutable requirement of `fn remove(&mut self)`.